### ISSUE TITLE
modules request thread

### ISSUE THREAD

**Issue by Eurikon** [bsp2/VeeSeeVSTRack#8]
Hope this is allowed to create an ongoing modules request thread here?
I understand that not all modules are possible to be supported. 

currently my most wanted missed modules inside VEESEEVEE are:
impromptuModular (all modules)
palmloop (21khz)
Southpole not parasites (all modules except the blanks)
waldemar (part of new lindberg research)
VCA-1 (fundamental)

thanks

**Comment by bsp2** [comment]
Sure, why not.

I've added your requested modules except for
- Southpole: I ported it a while ago but it crashed (don't have time to debug this, sorry)
 (this is the tweaked Clouds module, right ?)
- VCA-1: will look at this later

**Comment by Eurikon** [comment]
Hi man! thanks a lot!!

For Southpole i mean the list in the picture
![southpole](https://user-images.githubusercontent.com/40998416/44115513-60afe7e2-a00e-11e8-93d6-22774ef77481.png)
Anything you can get ported from this is very welcome! (except for the blanks).

Also missing Michael Struggl's modules: MSM
Frank Buss: Formula
Stellare Modular: Turing Machine

**Comment by bsp2** [comment]
The [latest release](https://github.com/bsp2/VeeSeeVSTRack/raw/v0.6/dist/veeseevstrack_0_6_1_win64_bin-16Aug2018.7z) now includes the
- Fundamental.VCA-1
 - Southpole.Abr
 - Southpole.Annuli
 - Southpole.Aux
 - Southpole.Balaclava
 - Southpole.Bandana
 - Southpole.Blank1HP
 - Southpole.Blank2HP
 - Southpole.Blank4HP
 - Southpole.Blank8HP
 - Southpole.Blank16HP
 - Southpole.Blank42HP
 - Southpole.But
 - Southpole.CornrowsX
 - Southpole.DeuxEtageres
 - Southpole.Etagere
 - Southpole.Falls
 - Southpole.Ftagn
 - Southpole.Fuse
 - Southpole.Gnome
 - Southpole.Piste
 - Southpole.Pulse
 - Southpole.Rakes
 - Southpole.Riemann
 - Southpole.Smoke
 - Southpole.Snake
 - Southpole.Sns
 - Southpole.Splash
 - Southpole.Sssh
 - Southpole.Wriggle

modules.

The MSM modules and the Turing Machine module are not open source so I cannot port them.

**Comment by Eurikon** [comment]
Geodesic :-) if they are open source i hope!

**Comment by bsp2** [comment]
[done](https://github.com/bsp2/VeeSeeVSTRack/raw/v0.6/dist/veeseevstrack_0_6_1_win64_bin-21Aug2018.7z) (the Pulsars module seems very suitable for re-creating the [Sounds of the Universe](https://www.youtube.com/watch?v=WE2fTDecEww) -- far out!)

**Comment by Eurikon** [comment]
thaaaaaaaaaaaaaaaaanks!!!

**Comment by Eurikon** [comment]
mental modules :)

**Comment by bsp2** [comment]
[done](https://github.com/bsp2/releases/raw/master/vsvr/veeseevstrack_0_6_1_win64_bin-25Aug2018.7z)

Keep in mind that some of the mental module UIs are tied to the audio engine, i.e. the buttons in the PatchMatrix module for example won't work while the plugin is idle.

**Comment by Eurikon** [comment]
ok thanks for pointing that out!

**Comment by Eurikon** [comment]
very essential update released for impromptu modular :)

**Comment by bsp2** [comment]
[done](https://github.com/bsp2/releases/raw/master/vsvr/veeseevstrack_0_6_1_win64_bin-28Aug2018.7z)

p.s.: the Impromptu sequencer UIs are also partially handled in the audio thread which means that they'll only work properly when the plugin is not idle

**Comment by Eurikon** [comment]
thanks for that!! 
and thanks for the heads-up, i didn't noticed the new Z function untill i needed it.
So I turn it always on for me, otherwise i can not receive CV sync through audio from the Silentway Sync plugin. 
I will have to take a look at the settings file to see if i can enable it by default.

**Comment by bsp2** [comment]
Setting "idleDetectInstr" / "idleDetectFx" to 0 (in settings.json) will turn off the idle detection for new instrument / fx instances.

**Comment by causticphoton** [comment]
I'd love to try the latest audible instruments: macro oscillator 2 (Plaits) and random sampler (Marbles)

**Comment by bsp2** [comment]
I've just merged them (Marbles / Plaits / Stages) into the repo. They'll be in the next binary release (probably in a few days).

**Comment by causticphoton** [comment]
> I've just merged them (Marbles / Plaits / Stages) into the repo. They'll be in the next binary release (probably in a few days).

Sweet! Thanks.

**Comment by davephillips** [comment]
Hi there ! :) Any chance we can get the SubmarineUtility and CharredDesert plugins ?

**Comment by bsp2** [comment]
Yes, write some plugins and share them with us ;)

Seriously, VSVR already comes with over 700(!) modules, and the release file sizes are now very close to the Github limit (98mb for the Linux release, 100mb is the limit).

Last but not least, I'm actually quite tired of porting modules (been doing that for 4 months now !).

There will always be new modules and / or module updates, and I can't keep on doing this forever (my spare time is limited, and I do have other projects to tend to !).

**Comment by davephillips** [comment]
Understood, no problem. I'll try my hand at porting a plugin or two, will let you know how I fare. Meanwhile, no worry about the SubUtility or CharredDesert stuff, I have quite enough to work with. :)

**Comment by cameronleger** [comment]
Is there a technical or licensing reason behind static vs. shared modules? I know that in the beginning, only static modules were possible, so it makes sense that most of them are that way. I'm open to porting and updating modules as well; without a solid reason for static modules I'd prefer to make them shared (easier for the user to add/remove as necessary).

**Comment by bsp2** [comment]
If I understand the GNU people correctly, the entire source needs to be GPL when one of the components is GPL, even if it's loaded dynamically. In case of VSVR: the Obxd module.
Since it's a shared module, users can simply delete the DLL / SO, and the VSVR plugin becomes MIT / BSD3 / CC again. A bit weird, isn't it :-)

The main reasons why most modules are linked statically are:
- dynamic modules are indeed rather new / came later
- due to all the `Rack_shared` library dependencies, each plugin has a certain minimum size  (~1.5MiB on Windows, ~2.5MiB on Linux)
   - i.e. if all 61 module sets were dynamic, the total binary size would be *much* larger, especially since there have to be two versions (instr+effect) of each plugin.
   - the only (good) solution would be to create an online plugin repository like the one in the standalone version of Rack, then download the plugins on-demand. I'm definitely not gonna do that.

Functionally, static and shared modules should be equal (especially after the FBO fix).
My own modules are loaded dynamically and I haven't found any issues with them so far.

**tl;dr**: shared modules are fine

**Comment by darloth** [comment]
Blamsoft's fairly high quality range of things are now free, if people hadn't noticed.

I had a quick look at the plausibility of including them as shared modules but didn't quite understand what/if anything was required to port them to VeeSeeVST rack format.

**Comment by kostiamisetskyy** [comment]
Hi there, i first have to thank you @bsp2 for the work you've done here, having such a powerful thing as vcv embedded in my daw is just mindbreaking !!

I have one module that i'm using everytime in vcv is the Turing machine from Stellare Modular 
(stellare-modular/vcv-rack repo on github) and i'll love to use it in my daw. i've seen you've put the skyline one but as far as i can see it's not as advanced and i play a lot with the "move" modulation and it's not there :( 

i have some basic knowledge in c++ so i've tried to build my version with this plugin but it gets out of my skills (or the configuration is wrong, tried several time, always fail...)
if i can build my own version with this plugin can you please give me some directions to do so ?
and if not can you please add it if i can ask ? (i'm running on windows btw)

One last question : i'm planning to use this on live performance, do you have any advices on long period stability with small patches ? (as far as i tested the vst module is far more stable with a daw beside that the standalone version, again this is mindbreaking !!! i can run 20 instances of vcv at the same time without any glitch or clock issue, BRAVO !!!)

**Comment by bsp2** [comment]
Re: Live performances: Two things I would be most concerned about:
1) Underruns. Depending on which DAW you are using, manually setting the priority of the DAW process to "realtime" (in the task manager or SysInternals Process Explorer) can reduce the risk of underruns considerably.
    You may also want to avoid instancing any plugins during your show, i.e. use the VSVR idle detection feature, use send effects (when possible), and preload all your plugins and sequences.
  
2) The "is (s)he checking his / her emails?" laptop effect. Try to control as many parameters as possible with MIDI controllers (the Allen&Heath XONE controllers are very nice), launch clips with a pad controller (or keyboard), maybe even use a multi channel interface, a (small) mixer and some guitar pedal fx to spice things up (the Elektron Analog Heat is very good for overdrive / distortion, and FMR produces some affordable but good sounding compressors, e.g. the RNC1773). Make sure to limit the MIDI controller ranges to sensible values. This can either be done in your DAW, or in VSVR take a look at the "Rescaler" and "Bias" modules. Depending on how many tracks you're going to play (which may require different controller setups), you may also want to create / print some overlays / labels for your MIDI controller(s) (i.e. to remind you of all the knob / slider / button functions).

Re: plugin stability: I've often kept the plugin running overnight and haven't seen any (long term) stability issues so far.
However, make sure to do a couple of "dry runs" to find any potential issues in your particular setup.

Re: Stellare Modular: Since this is not an open source plugin you have to ask the modular authors themselves for a VSVR port.

Last but not least: thanks for the kind words ! It's interesting to hear that you've found the VST to be more stable than the standalone version. Who would've thought !

**Comment by NRG70** [comment]
I wonder what people mean with "more stabile", I never had any stability issues with VCV Rack. As stability means crash for me. 
That said, performance wise VSVR might be performing better then VCV Rack. Probably cause the VST is running on multiple cores while loaded in the host DAW.

**Comment by ghost** [comment]
I like your plugins a lot, bsp2 that is, rms follower, use it everytime. Can you port it over to 1.0? Or is there something similair

**Comment by PtrMan** [comment]
You could split the common code in fx.so / .dll and instrument .so / .dll to reduce the filesize massivly.
